### ISSUE TITLE
Incompatibility in the URL parser with go 1.12.8 and github.com/go-sql-driver/mysql

### ISSUE THREAD

**Issue by aakn** [golang-migrate/migrate#264]
**Describe the Bug**

go-sql-driver has a DSN format that the go-1.12.8 does not consider as a valid URL.
Since the migrate tool parses the DSN before passing it on to the go-sql-driver, it leads to an error with the connection string.

**Steps to Reproduce**
1. Upgrade go to go-1.12.8
2. Rebuild the migrate tool
3. Run the command
    ```
    migrate -database "mysql://username:password@(127.0.0.1:3306)/database_name" -path db/migrations up
    ````
4. You'll see an unexpected error
    ```
    error: database: parse mysql://username:password@(127.0.0.1:3306)/database_name: invalid port ":3306)" after host
    ```

**Expected Behavior**
Migrations should've run successfully.

**Migrate Version**
e.g. v4.5.0

**Loaded Source Drivers**
file

**Loaded Database Drivers**
stub, mysql

**Go Version**
go version go1.12.8 linux/amd64

**Stacktrace**
NA

**Comment by lkebin** [comment]
ditto

**Comment by thaJeztah** [comment]
Reported the change in behavior in Golang (to confirm it was intentional); https://github.com/golang/go/issues/33646. I'll quote the response here;

https://github.com/golang/go/issues/33646#issuecomment-521349962

> Thanks for reporting this and sorry for the disruption. Indeed, this is unfortunate but I'm afraid necessary. DSNs like that are not well-formed URIs per RFC 3986, which is what net/url targets. All things being equal we will prefer backwards compatibility to standard compliance, but it became clear that allowing malformed ports was having a security impact on applications that assumed RFC 3986 compliance, so unless we break the world here we should stay on the stricter behavior.

> I mentioned this change in the golang-announce email, and had I noticed the test failure I would have opened an issue with github.com/go-sql-driver/mysql in advance. If somebody on this issue would like to open an issue there, I'd appreciate it, otherwise I'll look into which of their APIs are affected and let them know this evening.

**Comment by dhui** [comment]
We won't be able to cut a new release until the builds are fixed: https://github.com/golang-migrate/migrate/issues/254

**Comment by dnldaniel** [comment]
@dhui what about now? the issue does have impact to my team's software

**Comment by dhui** [comment]
Unfortunately, we don't have a centralized communications channel/procedure for issues and the latest status can only be gleaned from looking at the latest comments on related issues and PRs...

**Current status:**
An issue was reported with that the current fix (already merged into master) is insufficient.
I'm waiting for https://github.com/golang-migrate/migrate/pull/271 (the full fix) to be merged before cutting a new release.

If you believe that the fixes in master are sufficient for your needs, go ahead and use the master branch. If you'd like a more stable release, you'll need to wait for the next release. You can always use master for now and cut back to the next release when it's available.

@dnldaniel On a related note, if `migrate` is that important to your team, any help to improve it would be greatly appreciated. 😉

**Comment by dhui** [comment]
Fixed in [v4.6.0](https://github.com/golang-migrate/migrate/releases/tag/v4.6.0)
